R/labs_to_levels.R

```diff
@@ -0,0 +1,116 @@
+#' @title Convert value labels into factor levels
+#' @name labs_to_levels
+#'
+#' @details
+#' Convert variables or data into factors and uses value labels as factor levels.
+#' This function is similar to `to_factor()`, however, it is specially designed
+#' for factors with value labels attributes. While `to_factor()` returns factors
+#' as is, `labs_to_levels()` always uses value labels as factor levels, even
+#' for factors.
+#'
+#' @param x A data frame or vector.
+#' @param remove_attr Logical, if `"labels"` attribute should be removed after
+#' value labels are set as factor levels.
+#' @param ... Arguments passed to or from other methods, in this in particular
+#' to `to_factor()`.
+#' @inheritParams find_columns
+#' @inheritParams categorize
+#'
+#' @return A factor, or a data frame of factors.
+#'
+#' @examples
+#' str(labs_to_levels(iris))
+#'
+#' # use labels as levels
+#' data(efc)
+#' # create factor
+#' x <- as.factor(efc$c172code)
+#' # add value labels - these are not factor levels yet
+#' x <- add_labs(x, values = c(`1` = "low", `2` = "mid", `3` = "high"))
+#' levels(x)
+#' data_tabulate(x)
+#'
+#' x <- labs_to_levels(x)
+#' levels(x)
+#' data_tabulate(x)
+#' @export
+labs_to_levels <- function(x, ...) {
+  UseMethod("labs_to_levels")
+}
```

Comment:
For all methods, I think it should error if there are no labels

Reply:
I personally am not a big fan of error, but rather would use warnings. In our field, we often have data sets, which are not very clean, so some variables *may* have value labels while others do not. And if you want to iterate over a complete (large) data frame, it can be pain picking out and excluding those variables that don't work. These are rather rare occasions - but *if* it happens, users are mostly unable to deal with such errors (I teach R to absolute newcomers...)